src/Psalm/Config.php

```diff
@@ -2172,7 +2188,11 @@ public function getPredefinedConstants(): array
 
     public function collectPredefinedConstants(): void
     {
-        $this->predefined_constants = get_defined_constants();
+        $predefined_constants = get_defined_constants(true);
+        foreach (self::SUPPORTED_EXTENSIONS as $ext_name) {
+            unset($predefined_constants[$ext_name]);
```

Comment:
What's the point of that?

Reply:
Related to #7647, that's where constants that exist at runtime are loaded. This is basically removing any constants that come from a supported extension so that they'll instead be supplied by the stub.

Reply:
eg if you're running PHP 8.1 and have PDO enabled but you only require PHP "^8.0" this allows Psalm to warn you that [`PDO::FETCH_DEFAULT`](https://github.com/vimeo/psalm/pull/7641/files/8c4da1902a4134b2e5ce1c22a7af39f814f1d20e#diff-e91509795683dcfbba2e373ab4c4488dfce559643e06766ae954ccb19c2cad79R21-R22) doesn't exist.